test/Spec/HydraAuctionOnchain/Types/AuctionTerms.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Spec.HydraAuctionOnchain.Types.AuctionTerms
+  ( AuctionTerms (..)
+  , biddingPeriod
+  ) where
+
+import HydraAuctionOnchain.Types.AuctionTerms (PAuctionTerms)
+import Plutarch.DataRepr (DerivePConstantViaData (DerivePConstantViaData))
+import Plutarch.Lift (PConstantDecl, PUnsafeLiftDecl (PLifted))
+import PlutusLedgerApi.V2 (BuiltinByteString, POSIXTime, POSIXTimeRange, PubKeyHash, Value)
+import PlutusTx (makeIsDataIndexed)
+import Spec.HydraAuctionOnchain.Helpers (intervalFiniteClosedOpen)
+
+data AuctionTerms = AuctionTerms
+  { at'AuctionLot :: Value
+  , at'SellerPkh :: PubKeyHash
+  , at'SellerVk :: BuiltinByteString
+  , at'Delegates :: [PubKeyHash]
+  , at'BiddingStart :: POSIXTime
+  , at'BiddingEnd :: POSIXTime
+  , at'PurchaseDeadline :: POSIXTime
+  , at'Cleanup :: POSIXTime
+  , at'AuctionFeePerDelegate :: Integer
+  , at'StartingBid :: Integer
+  , at'MinBidIncrement :: Integer
+  , at'MinDepositAmount :: Integer
+  }
+  deriving stock (Show, Eq)
```

Comment:
Just out of curiosity: why do we need to redefine everything with PlutusTx types in the test suite?

Can Plutarch types not be used directly in test suites?

Reply:
Plutarch terms represent computations that may produce a value of a specified type if executed successfully, so they are effectively scripts that cannot be used as normal Haskell types without additional boilerplate.

Reply:
We can't use Plutarch types to build Haskell values, since they are just tags for computations that may produce a value. Plutarch terms, on the other hand, represent computations themselves, so `Term s MyType` will return a value of type `MyType` if executed successfully. These terms can in fact be converted to Haskell values using `plift`, which basically implies executing the computation (running the script), but in order to make sense of the result in the offchain code, there must be a corresponding Haskell type and `PUnsafeLiftDecl` instance.